src/models/transactions/accountSet.ts

```diff
@@ -0,0 +1,71 @@
+import { ValidationError } from "../../common/errors";
+import { BaseTransaction, verifyBaseTransaction } from "./common";
+
+enum AccountSetFlagEnum {
+    asfRequireDest = 1,
+    asfRequireAuth = 2,
+    asfDisallowXRP = 3,
+    asfDisableMaster = 4,
+    asfAccountTxnID = 5,
+    asfNoFreeze = 6,
+    asfGlobalFreeze = 7,
+    asfDefaultRipple = 8,
+    asfDepositAuth = 9,
+}
+
+export interface AccountSet extends BaseTransaction {
+    TransactionType: "AccountSet";
+    ClearFlag?: number;
+    Domain?: string;
+    EmailHash?: string;
+    MessageKey?: string;
+    SetFlag?: AccountSetFlagEnum;
+    TransferRate?: number;
+    TickSize?: number;
+    WalletLocator?: string;
+    WalletSize?: number;
+}
+
+/**
+ * Verify the form and type of an AccountSet at runtime.
+ * 
+ * @param tx - An AccountSet Transaction
+ * @returns - Void.
+ * @throws - When the AccountSet is Malformed.
+ */
+ export function verifyAccountSet(tx: AccountSet): void {
+    verifyBaseTransaction(tx)
+
+    if (tx.ClearFlag !== undefined){
+        if (typeof tx.ClearFlag !== 'number')
+            throw new ValidationError("AccountSet: invalid ClearFlag")
+        if (0 > tx.ClearFlag || tx.ClearFlag > 9)
+            throw new ValidationError("AccountSet: invalid ClearFlag")
+    }
+
+    if (tx.Domain !== undefined && typeof tx.Domain !== 'string')
+        throw new ValidationError("AccountSet: invalid Domain")
```

Comment:
I'm not a TypeScript expert but isn't verifying a type unnecessary when TS will catch this error at compile time?

Reply:
Yes, you are correct, in TS this type of runtime checking is unnecessary. We've been back and forth on whether or not to include this functionality. IMO I'd like to keep it + make it optional for the following reasons:
1. JavaScript users can use these `verify` functions
2. This functionality is similar to the jsonschemas functionality already in use in ripple-lib. This also lets us define exactly what error message we want to throw.
3. This functionality is pretty extendable to other runtime checks (something like an AccountDelete having a certain fee)
4. Typescript doesn't offer any runtime guarantees. Someone using TS wrong can still pass whatever they want to w/ an `any`.  Letting TS Clients choose to use this at runtime could be a useful feature.

I know @intelliot disagrees here, maybe he could write out his thoughts.

Reply:
That makes sense! I see no harm of keeping this optional for JS users.

I think it's worth documenting that we recommend users to call `verify` functions before submitting transactions; it would be helpful to describe this in jsdoc comments too. I also like the idea that it allows us to define specific error messages to throw.

@natenichols With that being said, I'm now supportive of this change and thanks for this detailed response! 😀

Reply:
> JavaScript users can use these verify functions

👍

> This functionality is similar to the jsonschemas functionality already in use in ripple-lib. This also lets us define exactly what error message we want to throw.

I don't find this functionality to be very useful. It's fine, but I don't really think it's worth the complexity + additional lines of code + need for maintenance.

> This functionality is pretty extendable to other runtime checks (something like an AccountDelete having a certain fee)

Note that the AccountDelete fee is equal to the owner reserve which can change based on a vote of the validators. So verifying that properly will require a Client (to retrieve the current owner reserve value).

> Typescript doesn't offer any runtime guarantees. Someone using TS wrong can still pass whatever they want to w/ an any. Letting TS Clients choose to use this at runtime could be a useful feature.

👍   But it happens rarely and early enough that I think it's also ok to get the error from rippled slightly later on.